### PR TITLE
new rabbitmq-k8s-diagnostic-run-book

### PR DESCRIPTION
**Pull request by DanielePalaia**
## Overview of changes

This is a new bash script to be intended as runbook for RabbiitMQ deployed on Kubernetes to implement this set of commands:
https://github.com/rabbitmq/support-tools/blob/main/docs/Reporting_RabbitMQ_For_Kubernetes_Issues.md

## Assessment of priority

Please delete the answers to the questions below as appropriate. 

For example, for the question: **How blocking is this documentation?:** The acceptable answers are: **Very/Somewhat/Not at all** if your answer is **Very** please delete the remaining answers **Somewhat** and **Not at all**.

**How blocking is this documentation?:** Not at all

**Is there a user waiting on this change?:** N

**Roughly how urgently is this documentation needed:** At some point

### CODE REVIEW

File: scripts/rabbitmq-k8s-runbook/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+./rabbitmq-k8s-diagnostic-runbook get_operators_info -n rabbitmq-system  ./diagnostics
+```
+
+This command returns informations and logs on the three RabbitMQ operators: cluster-operator, messaging-topology-operator and standby replication operator (this last one just available in the commercial version). </br>
+It takes in input the namespace where the operators are installed, if none is provided will search in the rabbitmq-system. 
+
+### Get RabbitMQ cluster diagnostic info:
+
+```
+./rabbitmq-k8s-diagnostic-runbook get_operators_info -n rabbitmq-cluster  ./diagnostics
```
Comment (coro) [inline scripts/rabbitmq-k8s-runbook/README.md]: I think this should be `get_rabbitmq_cluster_info` not `get_operators_info`

File: scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+
+
+    # secretgen-controller status namespace
+    printf "[INFO] Looking for secretgen-controller objects status and definitions in secretgen-controller namespace and saving it to files in $OUTPUT_DIRECTORY/carvel/info.\\n"
+    kubectl get all -n $SECRETGEN_NAMESPACE > $OUTPUT_DIRECTORY/carvel/info/sec_gen_get_all_objects_status.info
+    kubectl describe replicaset -n $SECRETGEN_NAMESPACE >> $OUTPUT_DIRECTORY/carvel/info/sec_gen_get_all_objects_status.info
+    kubectl describe deploy -n $SECRETGEN_NAMESPACE >> $OUTPUT_DIRECTORY/carvel/info/sec_gen_get_all_objects_status.info
+    kubectl get deploy -n $SECRETGEN_NAMESPACE -o yaml > $OUTPUT_DIRECTORY/carvel/info/secretgen-controller-definition.yaml
+
+    # secretgen-controller status namespace
```
Comment (coro) [inline scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook]: Should be packages namespace in the comment

File: scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+    printf "[INFO] Looking for PackageRepo and PackageInstall objects status in secretgen-controller namespace and saving it to files in $OUTPUT_DIRECTORY/carvel/info.\\n"
+    kubectl get PackageRepository -n $CARVEL_PACKAGES_NAMESPACE > $OUTPUT_DIRECTORY/carvel/info/packagerepo.info
+    kubectl describe PackageRepository -n $CARVEL_PACKAGES_NAMESPACE >> $OUTPUT_DIRECTORY/carvel/info/packagerepo.info
+    kubectl get PackageRepository -n $CARVEL_PACKAGES_NAMESPACE -o yaml > $OUTPUT_DIRECTORY/carvel/info/packagerepo_definition.yaml
+    kubectl get PackageInstall -n $CARVEL_PACKAGES_NAMESPACE > $OUTPUT_DIRECTORY/carvel/info/packageinstall.info
+    kubectl describe PackageInstall -n $CARVEL_PACKAGES_NAMESPACE >> $OUTPUT_DIRECTORY/carvel/info/packageinstall.info
+    kubectl get PackageInstall -n $CARVEL_PACKAGES_NAMESPACE -o yaml > $OUTPUT_DIRECTORY/carvel/info/packageinstall_definition.yaml
```
Comment (coro) [inline scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook]: I think these should fail if they don't exist - if we've specifically asked for carvel information and we don't get any then that should be raised as an error. In my tests I didn't notice I hadn't specified the right namespace.

File: scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+        usage
+        ;;
+    *)
+        usage
+        exit 1
+        ;;
+    esac
+}
+
+if [[ "$#" -ge 1 ]]; then
```
Comment (coro) [inline scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook]: Should this be 2? If I run without an output directory I just get:

```shell
$ ./rabbitmq-k8s-diagnostic-runbook get_rabbitmq_cluster_info
./rabbitmq-k8s-diagnostic-runbook: line 435: $1: unbound variable
```

where I feel the helptext would be valuable.

**Review by coro — APPROVED** [review]
A few comments but otherwise looks great to me! Thanks for this!

File: scripts/rabbitmq-k8s-runbook/README.md (hunk 1 above)
Comment (DanielePalaia) [reply scripts/rabbitmq-k8s-runbook/README.md]: Done!

File: scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook (hunk 2 above)
Comment (DanielePalaia) [reply scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook]: Done!

File: scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook (hunk 3 above)
Comment (DanielePalaia) [reply scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook]: Done! the script will now provide an ERROR log if PackageRepository or PackageInstall are not present in the provided namespace!

File: scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook (hunk 4 above)
Comment (DanielePalaia) [reply scripts/rabbitmq-k8s-runbook/rabbitmq-k8s-diagnostic-runbook]: Yes! Copied pasted from our plugin and forgot to adapt it :)

**Pull request merged**